### PR TITLE
Fix range argument in $ARISA_PURGE_ATTACHMENT

### LINKED ISSUE #759: `PurgeAttachmentCommand`'s range arguments don't work
## The Bug

The username argument for the purge attachment command is a greedy string, so the range arguments afterwards never get parsed.  For instance, `$ARISA_PURGE_ATTACHMENT Username 468330 468405` tries to purge all attachments from a user with the name `Username 468330 468405`, instead of trying to purge attachments from a user with the name `Username` where the attachment ID is between 468330 and 468405 (inclusive?).

https://github.com/mojira/arisa-kt/blob/b4be4c306d88ae7ede5208d68eb0f2fea40baca5/src/main/kotlin/io/github/mojira/arisa/modules/commands/CommandDispatcher.kt#L208

The test did not pick up on this, because the test directly runs the command instead of going through the parsing step.

## Example Ticket
https://bugs.mojang.com/browse/MC-247000 (I worked around this issue by purging all attachments and then re-uploading the relevant ones)

## Expected behavior
I'm not sure how to resolve this ambiguity while still having the range arguments be optional and the username argument be mandatory.

### PR DESCRIPTION
**Pull request by SPGoding**
## Purpose
Fix #759.

## Approach
By changing the syntax of the command from

```
$ARISA_PURGE_ATTACHMENT <username> [<minId> [<maxId>]]
```

to

```
$ARISA_PURGE_ATTACHMENT [from <minId>] [to <maxId>] [by <username>]
```

This makes all eight possible combinations of ways to filter attachments legal:

```
$ARISA_PURGE_ATTACHMENT                   // Remove all attachments from the report.
$ARISA_PURGE_ATTACHMENT by Annoying User  // Remove all attachments uploaded by "Annoying User" from the report.
$ARISA_PURGE_ATTACHMENT from 10000
$ARISA_PURGE_ATTACHMENT from 10000 by Annoying User
$ARISA_PURGE_ATTACHMENT from 10000 to 20000
$ARISA_PURGE_ATTACHMENT from 10000 to 20000 by Annoying User
$ARISA_PURGE_ATTACHMENT to 20000
$ARISA_PURGE_ATTACHMENT to 20000 by Annoying User
```

#### Checklist
- [x] Included tests
- [x] Updated documentation in [README](https://github.com/mojira/arisa-kt/blob/master/README.md) and [`docs` folder](https://github.com/mojira/arisa-kt/blob/master/docs)
- [ ] Tested in MCTEST-xxx

### CODE REVIEW

**Review by Marcono1234 — COMMENTED** [review]
Looks good! The command argument chain in `CommandDispatcher.kt` is a bit verbose, though maybe there is not an easier way to solve this.

**Comment by SPGoding** [comment]
Yeah thanks to Mojang/brigadier#19

File: docs/Commands.md [hunk 1]
```diff
@@ -105,18 +105,27 @@ the Security Level.
...
 
 ## $ARISA_PURGE_ATTACHMENT
-| Entry       | Value                                                      |
-| ----------- | ---------------------------------------------------------- |
-| Syntax      | `$ARISA_PURGE_ATTACHMENT <username> [<min ID>] [<max ID>]` |
-| Permissions | Mod+                                                       |
+| Entry       | Value                                                                   |
+| ----------- | ----------------------------------------------------------------------- |
+| Syntax      | `$ARISA_PURGE_ATTACHMENT [from <minId>] [to <maxId>] [by <username>]` |
+| Permissions | Mod+                                                                    |
```
Comment (Marcono1234) [inline docs/Commands.md:111]: Really minor, but the indentation does not seem to match:
```suggestion
| Entry       | Value                                                                 |
| ----------- | --------------------------------------------------------------------- |
| Syntax      | `$ARISA_PURGE_ATTACHMENT [from <minId>] [to <maxId>] [by <username>]` |
| Permissions | Mod+                                                                  |
```

(sorry forgot to add this comment before creating the review comment ...)

**Review by urielsalis — APPROVED** [review]

**Comment by Marcono1234** [comment]
@SPGoding, do you want to write in the internal communication channel to inform the others about the changed syntax?

**Pull request merged**